INA226.cpp

```diff
@@ -194,62 +194,97 @@ int INA226::setMaxCurrentShunt(float maxCurrent, float shunt, bool normalize)
     Serial.print("normalize:\t");
     Serial.println(normalize ? " true":" false");
     Serial.print("initial current_LSB:\t");
-    Serial.print(_current_LSB, 8);
+    Serial.print(_current_LSB * 1e+6, 1);
     Serial.println(" uA / bit");
   #endif
 
-  uint32_t calib  = 0;
-  uint32_t factor = 1;
-
   //  normalize the LSB to a round number
   //  LSB will increase
   if (normalize)
   {
-    calib = round(0.00512 / (_current_LSB * shunt));
-    _current_LSB = 0.00512 / (calib * shunt);
+    /* check if maxCurrent (normal) or shunt resistor (due to unusual low resistor values in relation to maxCurrent) determines currentLSB
+       we have to take the upper value for currentLSB
+    
+       calculation of currentLSB based on shunt resistor and calibration register limits (2 bytes)
+       cal = 0.00512 / ( shunt * currentLSB )
+       cal(max) = 2^16-1 
+       currentLSB(min) = 0.00512 / ( shunt * cal(max) )
+       currentLSB(min) ~= 0.00512 / ( shunt * 2^16 )
+       currentLSB(min) ~= 2^9 * 1e-5 / ( shunt * 2^16 )  
+       currentLSB(min) ~= 1e-5 / 2^7 / shunt
+       currentLSB(min) ~= 7.8125e-8 / shunt 
+    */
+    if(  7.8125e-8 / shunt > _current_LSB ) {
+      // shunt resistor determines currentLSB -> take this a starting point for currentLSB
+      _current_LSB = 7.8125e-8 / shunt;
+    }
 
     #ifdef printdebug
       Serial.print("Prescale current_LSB:\t");
-      Serial.print(_current_LSB, 8);
+      Serial.print(_current_LSB * 1e+6, 1);
       Serial.println(" uA / bit");
     #endif
 
-    //  auto scale current_LSB
-    factor = 1;
-    while (_current_LSB < 1)
-    {
-      _current_LSB *= 10;
-      factor *= 10;
+    // normalize _current_LSB to a value of 1, 2 or 5 * 1e-6 to 1e-3
+    // convert float to int
+    uint16_t currentLSB_uA = float(_current_LSB * 1e+6);
+    currentLSB_uA++; // ceil would be more precise, but uses 176 bytes of flash
+
+    uint16_t factor = 1;  // 1u to 1000u
+    uint8_t i = 0;        // 1 byte loop reduces footprint
```

Comment:
I prefer to have a separate bool fail flag as the value i is used for both iterator and status flag.